src/System.Data.SqlClient/tests/ManualTests/SQL/ExceptionTest/ExceptionTest.cs

```diff
@@ -217,6 +219,45 @@ public static void IndependentConnectionExceptionTest()
             }
         }
 
+        private static List<Exception> UnobservedTaskExceptionTest_Exceptions;
+
+        private static void TaskScheduler_UnobservedTaskException(object sender, UnobservedTaskExceptionEventArgs e)
+        {
+            Assert.False(UnobservedTaskExceptionTest_Exceptions.Contains(e.Exception.InnerException), "FAILED: This exception was already observed by awaiting: " + e.Exception.InnerException);
+        }
+
+        [CheckConnStrSetupFact]
+        public static async Task UnobservedTaskExceptionTest()
+        {
+            UnobservedTaskExceptionTest_Exceptions = new List<Exception>();
+            TaskScheduler.UnobservedTaskException += TaskScheduler_UnobservedTaskException;
+
+            using(var connection = new SqlConnection(DataTestUtility.TcpConnStr))
+            {
+                await connection.OpenAsync();
+                using (var command = new SqlCommand("select null; select * from dbo.NonexistentTable;", connection))
+                {
+                    try
+                    {
+                        using (var reader = await command.ExecuteReaderAsync())
+                        {
+                            do
+                            {
+                                while (await reader.ReadAsync())
+                                {
+                                }
+                            } while (await reader.NextResultAsync());
+                        }
+                    }
+                    catch (SqlException ex)
+                    {
+                        UnobservedTaskExceptionTest_Exceptions.Add(ex);
+                    }
+                }
+            }
+            GC.Collect();
```

Comment:
This should likely be:
```C#
GC.Collect();
GC.WaitForPendingFinalizers();
```
Otherwise the finalization that triggers the event may not happen during the test.